src/extension.ts

```diff
@@ -75,6 +77,46 @@ export async function activateInternal(context: ExtensionContext, perfStats: { l
 	return Object.assign(ext.loginHelper.legacyApi, createApiProvider([ext.loginHelper.api]));
 }
 
+async function checkAuthLibraryOnStartup(extensionContext: ExtensionContext, actionContext: IActionContext, loginHelper: AzureAccountLoginHelper): Promise<void> {
+	async function askThenSignOutAndReload(): Promise<void> {
+		const authLibraryChanged: string = localize('azure-account.authLibraryChanged', 'The authentication library has changed. Please sign out and reload the window for it to take effect.');
+		const signOutAndReload: string = localize('azure-account.signOutAndReload', 'Sign Out and Reload Window');
+		
+		// Purposefully await this message to block whatever command caused the extension to activate.
```

Comment:
Alternatively, should we _not_ await this, and just throw a `UserCancelledError`? What happens if they just hit the X?

Reply:
I vote we keep it as-is. I tried not awaiting this call and throwing a `UserCancelledError` immediately after and it continued on running the command anyway even after throwing the error.

How it currently is when we await the call: clicking X continues the command and results in the "Unexpected token..." error. But at that point I think it's the user's fault for ignoring our suggestion to sign out...

Reply:
Ah, yeah you're right. The call to the command does not contain the above in the call stack so throwing the `UserCancelledError` will throw out of activation, not out of the command execution. Nevermind!